### PR TITLE
Merge Channel Name feature to main

### PR DESCRIPTION
**Pull request by rakshith91**

### CODE REVIEW

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py [hunk 1]
```diff
@@ -187,12 +192,14 @@ def send(self, events, **kwargs):
         :keyword str content_type: The type of content to be used to send the events.
          Has default value "application/json; charset=utf-8" for EventGridEvents,
          with "cloudevents-batch+json" for CloudEvents
+        :keyword str channel_name: Optional. Used to specify the name of event channel when publishing to partner.
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/_publisher_client.py:195]: nit:
```suggestion
        :keyword channel_name: Optional. Used to specify the name of event channel when publishing to partner.
        :paramtype channel_name: str or None
```

File: sdk/eventgrid/azure-eventgrid/samples/async_samples/sample_publish_to_channel_async.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    1) EVENTGRID_PARTNER_NAMESPACE_TOPIC_KEY - The access key of your eventgrid account.
+    2) EVENTGRID_PARTNER_NAMESPACE_TOPIC_ENDPOINT - The topic hostname. Typically it exists in the format
+    "https://<YOUR-TOPIC-NAME>.<REGION-NAME>.eventgrid.azure.net/api/events".
+    3) EVENTGRID_PARTNER_CHANNEL_NAME - The name of the channel to which the event should be published.
+"""
+# [START publish_cloud_event_to_topic]
+import os
+import asyncio
+from azure.eventgrid.aio import EventGridPublisherClient
+from azure.core.credentials import AzureKeyCredential
```
Comment (swathipil) [inline sdk/eventgrid/azure-eventgrid/samples/async_samples/sample_publish_to_channel_async.py:22]: wondering if this needs to be an aio credential?

**Review by swathipil — APPROVED** [review]
left a few comments but o/w lgtm!

File: sdk/eventgrid/azure-eventgrid/samples/async_samples/sample_publish_to_channel_async.py (hunk 2 above)
Comment (rakshith91) [reply sdk/eventgrid/azure-eventgrid/samples/async_samples/sample_publish_to_channel_async.py:22]: nope - AzureKeyCredential doesn't have to be aio

File: sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py [hunk 3]
```diff
@@ -187,12 +187,14 @@ async def send(self, events: SendType, **kwargs: Any) -> None:
         :keyword str content_type: The type of content to be used to send the events.
          Has default value "application/json; charset=utf-8" for EventGridEvents,
          with "cloudevents-batch+json" for CloudEvents
+        :keyword str channel_name: Optional. Used to specify the name of event channel when publishing to partner
```
Comment (rakshith91) [inline sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_publisher_client_async.py:190]:
```suggestion
        :keyword channel_name: Optional. Used to specify the name of event channel when publishing to partner
        :paramtype channel_name: str or None
```

**Pull request merged**